plugins/engines/ospray/OSPRayScene.cpp

```diff
@@ -78,12 +78,12 @@ OSPRayScene::~OSPRayScene()
 void OSPRayScene::commit()
 {
     const bool rebuildScene = isModified();
+    const bool addRemoveVolumes = _commitVolumeData();
```

Comment:
Without knowing too much, I still find the logic here convoluted. I don't know why the isModified() is checked before _commitSimulationData and commitTransferFunctionData and what's so different with _commitVolumeData... Couldn't they just set the isModified() flag as needed, and then we would "ideally" have a commit() function that looks like this:
```
void OSPRayScene::commit()
{
    _commitVolumeData();
    _commitSimulationData();
    _commitTransferFunctionData();
    if (isModified())
        _rebuildScene();
}
```

Reply:
the modified state is also evaluated outside after this function to clear the framebuffer for instance. This happens if for instance the transfer function has changed, hence _commitTransferFunctionData() updates the modified state in the scene. But it does not require a rebuild of the scene, so the state is stored explicitly before.